ironfish/src/wallet/wallet.ts

```diff
@@ -1201,7 +1204,11 @@ export class Wallet {
       return
     }
 
-    await this.walletDb.cleanupDeletedAccounts(this.eventLoopAbortController.signal)
+    if (accountId) {
+      await this.walletDb.cleanupDeletedAccount(accountId, this.eventLoopAbortController.signal)
```

Comment:
This doesn't work as it does a mark and sweep amount of cleanup. Also these will now run in parallel with the event loop causing them to waste CPU and fight each other.

Reply:
Get it. Is there any other idea for this "wait for deletion finish" purpose?

Reply:
That's the entire feature request were asking to be built. If you cannot figure out how to implement it please don't pick up that feature request.